roles/translator/src/proxy/bridge.rs

```diff
@@ -104,7 +142,191 @@ impl Bridge {
+            loop {
+                let (mut message_new_template, token): (NewTemplate, u64) =
+                    new_template_reciver.recv().await.unwrap();
+                let (channel_id_to_new_job_msg, custom_job) = self_mutex
+                    .safe_lock(|a| {
+                        a.channel_factory.on_new_template(&mut message_new_template)
+                        // This is not efficient, to be changed if needed
+                    })
+                    .unwrap()
+                    .unwrap();
+                if let Some(custom_job) = custom_job {
+                    let req_id = self_mutex.safe_lock(|s| s.request_ids.next()).unwrap();
+                    let custom_mining_job = SetCustomMiningJob {
+                        channel_id: self_mutex
+                            .safe_lock(|s| s.channel_factory.get_this_channel_id())
+                            .unwrap(),
+                        request_id: req_id,
+                        version: custom_job.version,
+                        prev_hash: custom_job.prev_hash,
+                        min_ntime: custom_job.min_ntime,
+                        nbits: custom_job.nbits,
+                        coinbase_tx_version: custom_job.coinbase_tx_version,
+                        coinbase_prefix: custom_job.coinbase_prefix.clone(),
+                        coinbase_tx_input_n_sequence: custom_job.coinbase_tx_input_n_sequence,
+                        coinbase_tx_value_remaining: custom_job.coinbase_tx_value_remaining,
+                        coinbase_tx_outputs: custom_job.coinbase_tx_outputs.clone(),
+                        coinbase_tx_locktime: custom_job.coinbase_tx_locktime,
+                        merkle_path: custom_job.merkle_path,
+                        extranonce_size: custom_job.extranonce_size,
+                        future_job: message_new_template.future_template,
+                        token,
+                    };
+                    let tx_status = self_mutex.safe_lock(|s| s.tx_status.clone()).unwrap();
+                    handle_result!(tx_status, send_mining_job.send(custom_mining_job).await);
+                }
+
+                let (tx_sv1_notify, tx_status) = self_mutex
+                    .safe_lock(|s| (s.tx_sv1_notify.clone(), s.tx_status.clone()))
+                    .unwrap();
+                for (_id, job) in channel_id_to_new_job_msg {
+                    if let Mining::NewExtendedMiningJob(job) = job {
+                        handle_result!(
+                            tx_status.clone(),
+                            Self::handle_new_extended_mining_job_(
+                                self_mutex.clone(),
+                                &job,
+                                tx_sv1_notify.clone(),
+                            )
+                            .await
+                        )
+                    }
+                }
+                IS_NEW_TEMPLATE_HANDLED.store(true, std::sync::atomic::Ordering::SeqCst);
```

Comment:
What is the point of this one? It looks like there are 2 `IS_NEW_TEMPLATE_HANDLED` AtomicBools - this one starts as true and never gets checked and only gets set to true here. The one in the upstream is a separate instance which does seem to be used.